library/bignum.c

```diff
@@ -1328,7 +1328,8 @@ int mbedtls_mpi_add_abs( mbedtls_mpi *X, const mbedtls_mpi *A, const mbedtls_mpi
 }
 
 /*
- * Helper for mbedtls_mpi subtraction
+ * Helper for mbedtls_mpi subtraction:
+ * d -= s where d and s have the same size and d >= s.
```

Comment:
Not quite: `mpi_sub_hlp` works on `n` limbs of `s`, but goes beyond n limbs in `d` when it propagates the carry.

Reply:
I reworked the function to move the carry propagation out. Now it has a clearer semantics. It cost a few bytes of code size, but within the noise window.

```
$ git ss1; make --quiet -C build-m0plus && size build-m0plus/library/bignum.o
37ecc61 More logical parameter order for mpi_sub_hlp
  CC    source/library/bignum.c
  AR    library/libmbedcrypto.a
   text    data     bss     dec     hex filename
  15857       0       0   15857    3df1 build-m0plus/library/bignum.o
$ git ss1; make --quiet -C build-m0plus && size build-m0plus/library/bignum.o
c097e9e Move carry propagation out of mpi_sub_hlp
  CC    source/library/bignum.c
  AR    library/libmbedcrypto.a
   text    data     bss     dec     hex filename
  15869       0       0   15869    3dfd build-m0plus/library/bignum.o
$ git ss1; make --quiet -C build-m0plus && size build-m0plus/library/bignum.o
221626f Simplify the final reduction in mpi_montmul
  CC    source/library/bignum.c
  AR    library/libmbedcrypto.a
   text    data     bss     dec     hex filename
  15865       0       0   15865    3df9 build-m0plus/library/bignum.o
```